controllers/drplacementcontrol_controller.go

```diff
@@ -1343,6 +1343,17 @@ func (r *DRPlacementControlReconciler) finalizePlacement(
 	ctx context.Context,
 	placementObj client.Object,
 ) error {
+	// Ensure that OCM scheduling is disabled before we unprotect the application. If OCM scheduling
+	// is enabled, OCM may move the application to another cluster, losing currnet application data.
+	switch obj := placementObj.(type) {
+	case *clrapiv1beta1.Placement:
+		obj.Annotations[clrapiv1beta1.PlacementDisableAnnotation] = "true"
+	case *plrv1.PlacementRule:
+		obj.Spec.SchedulerName = RamenScheduler
```

Comment:
This changes the spec which we wanted to avoid, but there is no way to ensure that scheduling is disabled without changing the spec.

Reply:
OK, this can work, this will ensure that the placement[rule] is safe when we finish just like this change, and if it is not we can simply fail the function - this will log an big error with a traceback.

Reply:
After more investigation, this change is not needed. If the annotation is missing we will fail to get the placment[rule]. If we got to this point, the object is already valid and.

Even if we could validate, this is too late. Once you enable scheduling, OCM see the event and apply the placement to the app.